jetty-util/src/main/java/org/eclipse/jetty/util/ArrayTernaryTrie.java

```diff
@@ -96,10 +97,28 @@
      */
     private char _rows;
 
+    public static <V> AbstractTrie<V> from(int capacity, int maxCapacity, boolean caseSensitive, Set<Character> alphabet, Map<String, V> contents)
+    {
+        // Can't grow
+        if (maxCapacity < 0)
+            return null;
+        if (capacity < maxCapacity)
+            capacity = maxCapacity;
+        capacity++;
```

Comment:
That increment should be documented.

Reply:
It's no longer needed.  Removing

Reply:
It looks like some tests checking the exact capacity calculation might be needed...